src/data/finsupp.lean

```diff
@@ -68,11 +72,13 @@ lemma ext : ∀{f g : α →₀ β}, (∀a, f a = g a) → f = g
 lemma ext_iff {f g : α →₀ β} : f = g ↔ (∀a:α, f a = g a) :=
 ⟨by rintros rfl a; refl, ext⟩
 
-@[simp] lemma support_eq_empty [decidable_eq β] {f : α →₀ β} : f.support = ∅ ↔ f = 0 :=
+@[simp] lemma support_eq_empty {f : α →₀ β} : f.support = ∅ ↔ f = 0 :=
 ⟨assume h, ext $ assume a, by_contradiction $ λ H, (finset.ext.1 h a).1 $
   mem_support_iff.2 H, by rintro rfl; refl⟩
 
-instance [decidable_eq α] [decidable_eq β] : decidable_eq (α →₀ β) :=
+-- TODO Since this file is mostly classical anyway, it's unclear if this is useful.
+-- In any case, it's no longer an instance.
+def decidable_eq [decidable_eq α] [decidable_eq β] : decidable_eq (α →₀ β) :=
```

Comment:
I think this should stay an instance. It's not as relevant any more, but when you want decidability on finsupp this is the right way to get it. (Actually, the check can be simplified to `g.support ⊆ f.support ∧ (∀a∈f.support, f a = g a)`.)

Reply:
Okay, I've restored this. I didn't simplify the check, however.

Reply:
I'm struggling with this one. (For context, this PR was all compiling, until I tried rectifying this.)

I've turned it back into an instance, but now at [l.534 of `mv_polynomial.lean`](https://github.com/leanprover-community/mathlib/blob/decidable_eq/src/data/mv_polynomial.lean#L534) I get the usual
```
synthesized type class instance is not definitionally equal to expression inferred by typing rules, synthesized
  λ (a b : σ →₀ ℕ), finsupp.decidable_eq a b
inferred
  λ (a b : σ →₀ ℕ), classical.prop_decidable (a = b)
```
but I can't work out where to put a `convert` to fix it! Any help appreciated.

Reply:
Does it help to start the proof with `classical,`?

Reply:
Seem not too. :-(

Reply:
I pushed fixes to this and the other similar problems using `rw` (which uses unification and doesn't check that the instance comes out the same).